cmd/tailscale/cli/up.go

```diff
@@ -294,6 +296,29 @@ func runUp(ctx context.Context, args []string) error {
 		},
 	}
 
+	if upArgs.json {
+		opts.Notify = func(n ipn.Notify) {
+			if n.ErrMessage != nil {
+				fmt.Fprintf(os.Stdout, `{"error":%q}`, *n.ErrMessage)
+				os.Exit(1)
+			}
+			state := ""
+			if n.State != nil {
+				state = n.State.String()
+			}
+			if url := n.BrowseToURL; url != nil {
+				fmt.Fprintf(os.Stdout, `{"state":%q,"auth_url":%q}`, state, *url)
```

Comment:
but now it's without newlines.

I'd expect a JSON object per line. I think that's more common than concatenated.

Reply:
It only prints one JSON object then exits. This way a nice clean JSON object is written to stdout without trailing whitespace, which some parsers prefer.

Reply:
ah, relying on the cancel func.

Sure.

Last comment: `auth_url` style is at least the third JSON style we've introduced. We have GoCase, javascriptCase, and now underscore_case. Pick one of the first two instead of adding underscores?